spec/Section 6 -- Execution.md

```diff
@@ -552,9 +552,60 @@ set or coercing a scalar value.
 ExecuteField(objectType, objectValue, fieldType, fields, variableValues):
   * Let {field} be the first entry in {fields}.
   * Let {fieldName} be the field name of {field}.
+  * Let {requiredStatus} be the required status of {field}.
   * Let {argumentValues} be the result of {CoerceArgumentValues(objectType, field, variableValues)}
   * Let {resolvedValue} be {ResolveFieldValue(objectType, objectValue, fieldName, argumentValues)}.
-  * Return the result of {CompleteValue(fieldType, fields, resolvedValue, variableValues)}.
+  * Let {modifiedFieldType} be {ModifiedOutputType(fieldType, requiredStatus)}.
+  * Return the result of {CompleteValue(modifiedFieldType, fields, resolvedValue, variableValues)}.
+
+## Accounting For Client Controlled Nullability Designators
+
+A field can have its nullability status set either in its service's schema, or
+a nullability designator (! or ?) can override it for the duration of an execution.
+In order to determine a field's true nullability, both are taken into account
+and a final type is produced.
+
+ModifiedOutputType(outputType, requiredStatus):
+  * Create a {stack} initially containing {type}.
+  * As long as the top of {stack} is a list:
+    * Let {currentType} be the top item of {stack}.
+    * Push the {elementType} of {currentType} to the {stack}.
+  * If {requiredStatus} exists:
+    * Start visiting {node}s in {requiredStatus} and building up a {resultingType}:
```

Comment:
This section feels kind of clunky, but the only other part of GraphQL-js that uses `ASTReducer` is `print()` and that isn't part of the spec, so I didn't have anything to work off. I'm trying to describe a higher-order function here, but I don't know if it's clear. I'm super open to feedback!

Here's the implementation I was using as a reference: https://github.com/graphql/graphql-js/pull/3281/files#diff-5c5b9390cc439166260193ff061266d684b3ffce46b7b5d25588836c8b7c4b45R24